src/darsia/analysis/binarymassanalysis.py

```diff
@@ -0,0 +1,232 @@
+            base_image (darsia.Image): In principle only needed for size properties,
+                like numer of pixels, height and width.
+                depth_map (Optional[np.ndarray]): depth map with depth associated to
+                    each pixel.
+                depth_measurements (Optional[tuple[np.ndarray, ...]]): tuple of depth
+                    measurements. Should be provided as horizontal coordinates and
+                    vertical coordinates as a meshgrid, and depth cooridinates
+                    corresponding to each horizontal and vertical entry.
+                porosity (Optional[Union[np.ndarray, float]]): porosity map. Can be
+                    provided as a uniform scalar value as well.
+
+        """
+        self.base_image = base_image
+        self.pixelarea: float = self.base_image.dx * self.base_image.dy
+
+        # Define porosity map
+        if porosity is not None:
+            if isinstance(porosity, np.ndarray):
+                self.porosity: np.ndarray = porosity
+            else:
+                self.porosity = porosity * np.ones(self.base_image.img.shape[:2])
+        else:
+            self.porosity = np.ones(self.base_image.img.shape[:2])
+            warn("Please provide a porosity. Now it is assumed to be 1.")
+
+        # Define depth map
+        if depth_measurements is not None:
+            self.depth_map = self._compute_depth_map(depth_measurements)
+        elif depth_map is not None:
+            self.depth_map = depth_map
+        else:
+            warn(
+                "Please provide a depth_map (of at least the same size as "
+                " the segmented images that are to be analyzed) or a tuple of "
+                " depth measurements."
+                " Now, a constant depth of 1 is assumed."
+            )
+            self.depth_map = np.ones(self.base_image.img.shape[:2])
+
+        # Defines a height map associating each pixel with its physical
+        # height (distance from top)
+        self.height_map = np.linspace(
+            0, self.base_image.height, self.base_image.img.shape[0]
+        )[:, None] * np.ones_like(self.base_image.img)
+
+    def volume_map(self, segmentation: np.ndarray, component: int) -> np.ndarray:
+        """
+        Given segmentation and component, returns a volume map of the component
+        from the segmentation where each pixel gives the volume in meters^3 associated
+        to that pixel. If there is no compontent found at a pixel this map takes the value 0.
+
+        Arguments:
+            segmentation (np.ndarray): segmented image
+            component (int): the value that the component of interest takes in the
+                segmentation.
+
+        Returns:
+            volume map (np.ndarray): volume associated to each pixel containing the prescribed
+                component.
+        """
+
+        # Check that the depth map has the correct size.
+        if self.depth_map.shape != segmentation.shape:
```

Comment:
I think that we can remove the case where this goes through. I implemented it in the beginning as I though perhaps it might be convenient, but I think that it is always better to just recompute the depthmap for all separate cases. It is far to dangerous not to do it.

Reply:
I see points on both sides. I very much agree that sometimes recomputing an expensive field as the depth map seems tedious and an accelerated approach is desired. I usually used a resize in these cases, assuming that the dimensions of both situations are still somehow almost the same. To me resizing seems more suitable than cutting away data.